libs/controller/controller.py

```diff
@@ -157,34 +171,38 @@ def validate_equipment_availability(self, equipment_id=None):
 
     # Need to be added in future
     def request_ap_reboot(self):
+        self.refresh_instance()
         pass
 
     # Get the equipment id, of a equipment with a serial number
     def get_equipment_id(self, serial_number=None):
+        self.refresh_instance()
         equipment_data = self.get_equipment_by_customer_id(max_items=100)
-        # print(equipment_data)
+        # print(len(equipment_data))
         for equipment in equipment_data:
-            print(equipment._id)
             if equipment._serial == serial_number:
                 return equipment._id
 
     # Get the equipment model name of a given equipment_id
     def get_model_name(self, equipment_id=None):
+        self.refresh_instance()
         if equipment_id is None:
             return None
+        self.refresh_instance()
         data = self.equipment_client.get_equipment_by_id(equipment_id=equipment_id)
         print(str(data._details._equipment_model))
         return str(data._details._equipment_model)
 
     # Needs Bug fix from swagger code generation side
     def get_ap_firmware_new_method(self, equipment_id=None):
-
+        self.refresh_instance()
```

Comment:
so refresh function is called in every controller lib function to ensure we never runs any controller api if token is expired. The behavior of this function is that it will only refresh if timeout happened. then it will reset the timeout